indico/modules/rb/client/js/common/rooms/edit/RoomEditOptions.jsx

```diff
@@ -0,0 +1,150 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2020 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import React, {useMemo} from 'react';
+import {Dropdown, Form, Header, Tab} from 'semantic-ui-react';
+import _ from 'lodash';
+import {FieldArray} from 'react-final-form-arrays';
+import PropTypes from 'prop-types';
+import {Translate} from 'indico/react/i18n';
+import {FinalCheckbox, FinalField, FinalInput, validators as v} from 'indico/react/forms';
+import EquipmentList from './EquipmentList';
+import DailyAvailability from './DailyAvailability';
+import NonBookablePeriods from './NonBookablePeriods';
+
+export default function RoomEditOptions({active, showEquipment, globalAttributes}) {
+  const attributeTitles = useMemo(
+    () => (globalAttributes ? _.fromPairs(globalAttributes.map(x => [x.name, x.title])) : []),
+    [globalAttributes]
+  );
+
+  return (
+    <Tab.Pane active={active}>
+      <Header>
+        <Translate>Availability</Translate>
+      </Header>
+      <Form.Group>
+        <FinalField name="bookable_hours" component={DailyAvailability} isEqual={_.isEqual} />
+        <FinalField name="nonbookable_periods" component={NonBookablePeriods} isEqual={_.isEqual} />
+      </Form.Group>
+      <Header>
+        <Translate>Equipment and custom attributes</Translate>
+      </Header>
+      <Form.Group>
+        {showEquipment && (
+          <FinalField
+            name="available_equipment"
+            component={EquipmentList}
+            isEqual={_.isEqual}
+            componentLabel={Translate.string('Add new equipment')}
+          />
+        )}
+        {globalAttributes && globalAttributes.length > 0 && (
+          <FieldArray name="attributes" isEqual={_.isEqual}>
+            {({fields}) => {
+              if (!fields.value) {
+                return null;
+              }
+              const fieldsByName = new Set(fields.value.map(x => x.name));
```

Comment:
Why `Set` here? Don't you get unique values here in `fields`?

Reply:
Suggested [here](https://github.com/indico/indico/pull/4408/files/088a9cfcce9e3cb2f544c55be8132e71b6560bdc#diff-cd3d7e574e8639b08a5920392ad4b9ba) as a minor improvement for the `includes` check below.